### ISSUE TITLE
Add `--show-recent-failures`

### ISSUE THREAD

**Issue by zbraniecki** [mozilla/sccache#624]
When working with sccache it's quite painful to debug the failures because one needs to restart the server in the debug mode and filter out tons of debug messages to find the failures.

It would be useful to be able to do `sccache --show-recent-failures` and get the dump of just failures as reported in stats.

It likely depends on #33.

**Comment by froydnj** [comment]
Quite painful to debug which failures?  Figuring out why particular things didn't get cached and/or distributed?

**Comment by zbraniecki** [comment]
Yes. I just finished a compilation of Firefox and I see in my stats:

```
Failed distributed compilations      25
```

but I don't have a good way to see what failed and why.
